Fix: Copy object mtime

N-o-Z opened the pull request:
Closes #8290

yonipeleg33 commented on pkg/catalog/catalog.go:
```diff
@@ -2741,6 +2740,9 @@ func (c *Catalog) CopyEntry(ctx context.Context, srcRepository, srcRef, srcPath,
 		return nil, err
 	}
 
+	// Update creation date only after actual copy!!!
```
```suggestion
	// Update creation date only after actual copy to prevent potential discrepancy between lakefs and the underlying blockstore mtimes
```

N-o-Z replied:
This comment is incorrect, it does not prevent any discrepancy

yonipeleg33 replied:
Huh, this is kind of what I understood from reading the issue description. Maybe "prevent" isn't accurate, as this fix aims to minimise the time difference rather than making it equal, but maybe I got it all wrong 😄
In any case, my point is to add a _why_ for this comment, not only _what_.

yonipeleg33 commented on pkg/catalog/catalog.go:
```diff
@@ -2741,6 +2740,9 @@ func (c *Catalog) CopyEntry(ctx context.Context, srcRepository, srcRef, srcPath,
 	}
 
+	// Update creation date only after actual copy!!!
+	dstEntry.CreationDate = time.Now()
```
I wonder - shouldn't we use the blockstore's mtime instead of `time.Now`? Is this even possible?

N-o-Z replied:
Out of scope for this fix. Your suggestion is a valid question which has implications that we are currently  evaluating since it requires another call to the underlying storage

yonipeleg33 reviewed (commented):
Thanks!
Not approving, but I do not wish to block this PR either.
Also, please change the title to "Fix: change object mtime after blockstore copy" or something a bit more indicative

N-o-Z commented:
> Approving with couple of suggestions:
> 
> * Include `CompletePresignMultipartUpload` & `StageObject` similar fixes in this PR (or at the very least open an issue for the same phenomena)
> * If there's some easy way to test it I'd add that timestamp validation there. We know it causes bug for users, so it's best to try and prevent regressions.

@itaiad200 StageObject is irrelevant since no writes to the storage are being performed. Modified the code in `CompletePresignMultipartUpload` though in reality it's not an issue.
Added tests

N-o-Z commented:
Gave up on adding tests - too complicated and does not justify the overhead. Tested manually
